app/classifier/classifier.jsx

```diff
@@ -147,14 +150,18 @@ class Classifier extends React.Component {
     this.props.classification.update(changes);
   }
 
+  handleModelScoreUpdate(newScore) {
+    this.props.workflow.configuration.metadata.modelScore = newScore;
```

Comment:
This looks dangerous to me. Volunteers shouldn't be modifying the workflow configuration.

If the intention here is to update the state of the score, either update component state or update the score as part of the task annotation.

Reply:
I tried updating component state, but it was a huge performance hit (not sure why) so not feasible. 
Where in the annotation would you recommend? Would `classification.metadata` be safe enough? I don't know what gets sent back to the api.

Reply:
The annotation will be in `classification.annotations` and I think the annotation for the current task is passed down as a prop, so maybe `this.props.annotation`. The actual structure of the annotation varies from task to task, so depends on what sort of task modelling is. I'm guessing maybe some sort of drawing task?

Reply:
The only workflow so far is series of combo tasks with drawing and sliders, but I could potentially see just a drawing task being used.
It's the kind of thing that sits perfectly on state, I'll try and debug the performance problems. I think it's just triggering a huge number of re-renders.

Reply:
This sounds familiar. @rogerhutchings might be helpful here because he ran into a similar problem with the freehand drawing tool triggering hundreds, if not thousands, of re-renders when you drew a line. I think his solution was to buffer changes until drawing was complete, then trigger an annotation change, which in turn triggers a re-render of the classifier.

Reply:
React's `shouldComponentUpdate` is helpful in stopping unnecessary re-renders. Not sure if it would help in this case.

Reply:
I know @rogerhutchings is a big fan of that method, so that might be what he used.